app/models/health_care_application.rb

```diff
@@ -218,6 +236,30 @@ def parsed_form
     @parsed_form ||= form.present? ? JSON.parse(form) : nil
   end
 
+  def send_event_bus_event(status, next_id = nil)
+    return unless Flipper.enabled?(:hca_kafka_submission_enabled)
+
+    Rails.logger.info '~~~~~~~~~~~~~~~ send_event_bus_event', status, next_id
+
+    begin
+      user_icn = user&.icn || self.class.user_icn(self.class.user_attributes(parsed_form))
+      # local testing
+      # user_icn = user&.icn || '12345678'
+    rescue Common::Exceptions::ValidationErrors
+      # if certain user attributes are missing, we can't get an ICN
+      user_icn = ''
```

Comment:
I'm not too familiar with how Avro schema works but if a field is nullable, can it be omitted? in the `submit_event` the next_id is being omitted if there's no value. May be helpful to do the same with icn or at least set it to nil.

Reply:
The `icn` is a required argument in the call to `Kafka.submit_event`. As part of the validation ticket we might want to change that to make it more consistent. For now, I set it to nil rather than an empty string.